Feature/task 5 implement import functi

JAmmermann-DLR opened the pull request:
Add import functionality:
- Add JSON file selection and full importing that file to the import command
- Add full import functionality
- Add inheritance
- Add testcases for full import

CRUD will be tested later in https://github.com/virtualsatellite/VirtualSatellite4-FreeCAD-mod/issues/18

Closes #5

Commit:
Update commands and add file open dialog to import command - (Task #5)

- Fixed naming in the dialogs
- Switched command positions in InitGui
- Added file open dialog with JSON filter to import command

---
Task #5: Implement Import Functionality

Commit:
Add import functionality to CommandImport - (Task #5)

Use the JsonImporter in the CommandImport

---
Task #5: Implement Import Functionality

Commit:
Refactor importer and add fixture - (Task #5)

- Add full import function in JsonImporter
- Use that function in CommandImport
- Add test function for full import
- Add test fixture json file

---
Task #5: Implement Import Functionality

Commit:
Call JsonProductAssembly in full_import function- (Task #5)

- Add JsonProductAssembly call
- Add json product tests

---
Task #5: Implement Import Functionality

Commit:
Add inheritance to JsonProductChild - (Task #5)

Also:
- Add test data in test_json_data
- Add test case with that data in test_json_product_assembly
- Update test_full_import

---
Task #5: Implement Import Functionality

Commit:
Add functionality for importing a JSON twice - (Task #5)

Importing twice should result in the same output as importing once.
json_importer: Create FreeCAD filename from JSON filename
test_json_importer: Add testcase for importing twice
json_product:
- Add function to compare AJsonProducts
- Only create a part if it doesn't exist already
- Clear the placement when reimporting

---
Task #5: Implement Import Functionality

Commit:
Add position and rotation propagation - (Task #5)

Add a simple solution in json_product_child to propagate values from
parent to children and also test this behavior.

The current method results in children only having an absolute position
(and rotation) instead of a relative one to their parent. This should be
reconsidered with usability and the export functionality in mind.

---
Task #5: Implement Import Functionality

Commit:
Add functionality for full import twice with changes - (Task #5)

Because of CRUD assume that a change from Virtual Satellite side has
always newer information, so we can simply clear the first imported file
and use the newer file. For that:

- Add a function to clear a file in active_document
- Call that function in full_import
- Get FreeCAD document from ProductRoot instead of filename (so
different versions can easily be saved in different files)
- Add new test fixture

---
Task #5: Implement Import Functionality

Commit:
Resolve TODOs - (Task #5)

- Add a module directory in FreeCAD Appdata:
	- Create directory in Init
	- Provide it with Environment
	- Use it in CommandImport
- Update documentation and Log Messages
- Update part update in json_product: Now deletes and recreates existing
parts

---
Task #5: Implement Import Functionality

Commit:
Fix missing Mod directory in AppData - (Task #5)

Also add the Mod directory

---
Task #5: Implement Import Functionality

PhilMFischer commented on VirtualSatelliteCAD/commands/command_import.py:
```diff
@@ -28,19 +28,35 @@
+        if filename != '':
+            FreeCAD.Console.PrintMessage(f"Selected file '{filename}'\n")
+
+            json_importer = JsonImporter(Environment.get_appdata_module_path() + os.sep)
```
Why do we have to hand over this appdata_module_path?

JAmmermann-DLR replied:
Created an issue for this: https://github.com/virtualsatellite/VirtualSatellite4-FreeCAD-mod/issues/19

Commit:
Disable full import test cases and add TODOs - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Add test_json_product_assembly test cases - (Task #5)

Add:
- test_create_part_product_subassembly_with_root_part
- test_create_part_product_assembly_and_subassembly_with_root_part
- test_create_part_product_assembly_with_root_part_with_traverser (empty
test)

---
Task #5: Implement Import Functionality

Commit:
Add JsonProductAssemblyTreeTraverser - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Add test_json_product_assemby_tree_traverser - (Task #5)

- Add first test cases
- Add the test class to TestVirtualSatelliteApp
- Update traverser
- Update product assembly test
---
Task #5: Implement Import Functionality

Commit:
Cleanup traverser - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Use traverser in importer and add test cases- (Task #5)

Try to use the traverser in the importer and reactivating the
full_import test case showed problems with assemblies having the same
name and uuid as their part.
To further investigate that, test cases were added in the traverser.

---
Task #5: Implement Import Functionality

Commit:
Add identifiers for product assembly and part files - (Task #5)

These solve the problem of having a part and an assembly with the same
name and uuid.

- Add the identifiers to json_definitions
- Update the names in part and product files and corresponding test
files

---
Task #5: Implement Import Functionality

Commit:
Reactivate full_import test case and update importer - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Refactor command_import and resolve TODOs - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Remove unused import and blank lines at EOF - (Task #5)

---
Task #5: Implement Import Functionality

Commit:
Merge branch 'development' into feature/task_5-Implement_Import_Functi
